### DISCUSSION TITLE
[Feature] 请求支持自动检测并通过cloudflare5秒盾

### DISCUSSION THREAD

**Discussion by sunjx97** [qd-today/qd#282] [Ideas]
### Verify steps

- [X] Tracker 我已经在 [Issue Tracker](……/) 中找过我要提出的问题
- [X] Need 当前 QianDao 框架并不包含该功能特性或者还不完善
- [X] Framework 这是 QianDao 框架应包含的特性，并非模板特性
- [X] Meaningful 我提交的不是无意义的 催促更新或修复 请求

### Describe the Feature

有部分站点经常开启cloudflare5秒盾，导致签到失败。

### Describe the solution

希望能增加自动检测并通过cloudflare5秒盾，最好能内置，感谢

### Describe alternatives

通过API通过5秒盾，如FlareSolverr（配置FlareSolverr API URL），但听说效果并不好orz

**Comment by LD-KiLLer** [comment]
cloudflare的5秒盾无解，但是可以通过修改签到文件里的User-Agent匹配你的Cookie来解决

**Comment by hunter8605** [comment]
> cloudflare的5秒盾无解，但是可以通过修改签到文件里的User-Agent匹配你的Cookie来解决

修改了UA，还是不行。用浏览器打开网站一遍，就可以签到，第二天又不行了，不知道是什么原因

**Comment by caspershw** [comment]
https://github.com/Anorov/cloudflare-scrape
这个库应该有用，但是可能无法很好实现，并且还需要nodejs依赖

**Comment by yaohappiness** [comment]
有些站点安全性较高，很容易触发 cloudflare 5 秒盾，不只是 User-Agent 的原因。

> https://github.com/Anorov/cloudflare-scrape 这个库应该有用，但是可能无法很好实现，并且还需要 nodejs 依赖

上面的库已经失效了，有其他替代的方案：[Is cfscrape dead? · Issue #406 · Anorov/cloudflare-scrape](https://github.com/Anorov/cloudflare-scrape/issues/406#issuecomment-754799513)

**Comment by a76yyyy** [comment]
事实上, 不管是什么解决方案, 都需要时刻跟随cloudflare的更新节奏。 目前已知的所有解决方案似乎都不能一劳永逸或者时刻跟随这样的猫鼠游戏节奏， 而框架本身的目的也不是为了这样的猫鼠游戏而去做频繁地代码迭代， 因此目前暂时将该 issue 迁移至 discussion 讨论区，并暂时不考虑开发此功能

**Reply by sunjx97** [reply]
嗯...感觉还是加一个这样的实验性功能吧，就直接用上面别人的库，然后docker环境变量控制开启。
加了还有可能有用，没有就彻底没办法了（

**Reply by a76yyyy** [reply]
目前模板库有很多使用 `flaresolverr` 的模板，你可以参考一下，这些模板的应用证明框架确实无需内置过cf的功能

**Comment by github-h** [comment]
你可以关注一下  [FlareSolverr/FlareSolverr](https://github.com/FlareSolverr/FlareSolverr) 这个项目
